mypy/errors.py

```diff
@@ -375,12 +396,48 @@ def messages(self) -> List[str]:
             a.append(s)
         return a
 
+    def new_file_messages(self, path: str) -> List[str]:
+        """Return a string list of new error messages from a given file.
+
+        Use a form suitable for displaying to the user.
+        Formatted messages are cached in the order they are generated
+        by new_file_messages() in order to have consistency in output
+        between incrementally generated messages and .messages() calls.
+        """
+        if path not in self.error_info_map:
+            return []
+        msgs = self.format_messages(self.error_info_map[path][self.new_errors_start_map[path]:])
+        self.new_errors_start_map[path] = len(self.error_info_map[path])
+        self.formatted_messages += msgs
+        return msgs
+
+    def new_messages(self) -> List[str]:
+        """Return a string list of new error messages.
+
+        Use a form suitable for displaying to the user.
+        Errors from different files are ordered based on the order in which
+        they first generated an error.
+        """
+        msgs = []
+        for key in self.error_info_map.keys():
+            msgs.extend(self.new_file_messages(key))
+        return msgs
+
+    def messages(self) -> List[str]:
+        """Return a string list that represents the error messages.
+
+        Use a form suitable for displaying to the user.
+        """
+        self.new_messages()
```

Comment:
Add comment about `new_messages()` storing new messages as a side effect (or potentially rename the method to something that makes this more explicit).